resources/ensure_rate_limiting_setup.lua

```diff
@@ -0,0 +1,18 @@
+local rlKey = KEYS[1]
+local tokenKey = KEYS[2]
+local epochKey = KEYS[3]
+local currentEpoch = ARGV[1]
+local rateLimit = ARGV[2]
+local keyTimeout = ARGV[3]
+
+if redis.call('get', rlKey) == false then
+    redis.call('setex', rlKey, keyTimeout, rateLimit)
+end
+
+if redis.call('get', tokenKey) == false then
+    redis.call('setex', tokenKey, keyTimeout, rateLimit)
+end
+
+if redis.call('get', epochKey) == false then
+    redis.call('setex', epochKey, keyTimeout, currentEpoch)
```

Comment:
you cannot use NX/XX because expiry can only be set with Setex.  You could potentially do SETNX -> expire.  Which might be better.  But other than round trip time, there seems to be no difference between doing this in this file vs doing it all in luascript.  I like separating them out for the  readability / understandability.